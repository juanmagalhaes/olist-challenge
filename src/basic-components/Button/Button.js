// @flow

import * as React from 'react';
import classNames from 'classnames';
import styles from './Button.module.scss';

type Props = {
  color: string,
  children: React.Node,
}

const Button = ({ color, onClick, children, ...rest }: Props) => (
  <button
    {...rest}
    className={classNames(
      styles.button,
      { [styles.buttonSuccess]: color === 'success' }
    )}
    onClick={evt => {
      // add click animation
      evt.persist();
      const { currentTarget } = evt;
      currentTarget.classList.add(styles.buttonClicked);
      setTimeout(
        () => currentTarget.classList.remove(styles.buttonClicked),
        1000
      );
    }}
  >
    {children}
  </button>
);

export { Button };

