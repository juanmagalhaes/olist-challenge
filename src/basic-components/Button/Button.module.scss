@import '~global-styles/variables';

$button-width: 375px;
$button-height: 52px;
$button-font-size: 18px;
$button-font-weight: bold;
$button-margin: 8px 0 0;

@keyframes ripple {
  0% {
    transform: scale(0, 0);
    opacity: 1;
  }
  20% {
    transform: scale(40, 40);
    opacity: 1;
  }
  100% {
    opacity: 0;
    transform: scale(70, 70);
  }
}

.button {
  border: none;
  cursor: pointer;
  width: $button-width;
  margin: $button-margin;
  height: $button-height;
  font-size: $button-font-size;
  font-weight: $button-font-weight;
  position: relative;
  overflow: hidden;

  &:after {
    content: '';
    position: absolute;
    top: 50%;
    left: 50%;
    width: 5px;
    height: 5px;
    background: rgba(255, 255, 255, .5);
    opacity: 0;
    border-radius: 100%;
    transform: scale(1, 1) translate(-50%);
    transform-origin: 50% 50%;
  }

  &Clicked {
    &:focus:not(:active)::after {
      animation: ripple 1s ease-out;
    }
  }
}

.buttonSuccess {
  color:$success-cl;
  background: $success-bg;
  &:disabled {
    cursor: not-allowed;
  }
}

