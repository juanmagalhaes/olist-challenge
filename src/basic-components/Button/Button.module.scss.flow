// @flow
/* This file is automatically generated by css-modules-flow-types */
declare module.exports: {|
  +'button': string;
  +'buttonClicked': string;
  +'buttonSuccess': string;
  +'ripple': string;
|};
