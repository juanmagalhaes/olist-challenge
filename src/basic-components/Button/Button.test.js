import React from 'react';
import { Button } from '../Button';

test('Button', () => expect(shallow(<Button />)).toMatchSnapshot());
test('Button success', () => expect(shallow(<Button color="success" />)).toMatchSnapshot());
test('Button with text', () => expect(shallow(<Button>Submit</Button>)).toMatchSnapshot());

