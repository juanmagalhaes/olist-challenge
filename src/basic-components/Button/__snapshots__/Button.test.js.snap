// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 1`] = `
<button
  className="button"
  onClick={[Function]}
/>
`;

exports[`Button success 1`] = `
<button
  className="button buttonSuccess"
  onClick={[Function]}
/>
`;

exports[`Button with text 1`] = `
<button
  className="button"
  onClick={[Function]}
>
  Submit
</button>
`;
